pydra/engine/tests/test_boutiques.py

```diff
@@ -26,12 +26,12 @@
     "maskfile", ["test_brain.nii.gz", "test_brain", "test_brain.nii"]
 )
 @pytest.mark.parametrize("results_function", [result_no_submitter, result_submitter])
-def test_boutiques_1(maskfile, plugin, results_function):
+def test_boutiques_1(maskfile, plugin, results_function, tmpdir):
     """ simple task to run fsl.bet using BoshTask"""
     btask = BoshTask(name="NA", zenodo_id="1482743")
     btask.inputs.infile = Infile
     btask.inputs.maskfile = maskfile
-    res = results_function(btask, plugin)
+    res = results_function(btask, plugin, tmpdir)
```

Comment:
It will be better if you don't modify the `result_function`, just add `tmpdir` to the `btask` directly

Reply:
you actually add `cache_dir` only to one of the function that is used as `results_function`

Reply:
Ah ok. I thought it's just more convenient and it's good practice to use tmpdir in general.  Though I can go back and modify the code again

Reply:
As you could see `result_no_submitter` and `result_submitter` are really minimal functions that were design to run the task and return the result. I don't want to add anything extra (plus you only modified one of them).

And I don't really see any gain, since you have to change every test anyway.

Reply:
> you actually add `cache_dir` only to one of the function that is used as `results_function`

Not sure if I catch what you mean here?

Reply:
`results_function` is a variable used in the `parametrize` field and it could be either `result_no_submitter` or `result_submitter`. You added `tmpdir`  to `results_function` that suggests that this is used in both  `result_no_submitter` and `result_submitter`, but this is not the case, you only modified `result_submitter`.

But I'd really keep these functions as the minimal functions that run the task either using `Submitter` explicitly or not. Everything else can be set only once, before calling `results_function`